test/distributed/tensor/parallel/test_2d_parallel.py

```diff
@@ -77,9 +97,16 @@ def init_model(model_parallel_size=TP_DEGREE, use_orig_params=False, fsdp_nested
 
     # Create Input
     model = _distribute_and_fsdp_wrap_module(
-        model, True, twod_mesh, fsdp_pg, use_orig_params, fsdp_nested
+        model,
+        True,
+        twod_mesh,
+        fsdp_pg,
+        use_orig_params,
+        fsdp_nested,
+        recompute_activation,
     )
-    return model, fsdp_pg
+    tp_mesh = _create_1d_device_mesh(twod_mesh, 1)
```

Comment:
nit: This sort of mesh manipulation should be an integral part of DeviceMesh itself.

Reply:
Yes I totally agree but it's outside the scope of this PR. I can send a follow-up PR to address this or someone is already working on it?

Reply:
A follow up PR would be beautiful!